internal/pkg/service/buffer/store/cleanup.go

```diff
@@ -0,0 +1,74 @@
+package store
+
+import (
+	"context"
+	"time"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/filestate"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/slicestate"
+	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
+)
+
+// Cleanup deletes all unneeded tasks, files and slices data from the store.
+func (s *Store) Cleanup(ctx context.Context, receiver model.Receiver) (err error) {
+	_, span := s.tracer.Start(ctx, "keboola.go.buffer.store.Cleanup")
+	defer telemetry.EndSpan(span, &err)
+
+	now := time.Now()
+
+	// Delete all old tasks.
+	tasks, err := s.schema.Tasks().InReceiver(receiver.ReceiverKey).GetAll().Do(ctx, s.client).All()
+	if err != nil {
+		return err
+	}
+	for _, t := range tasks {
+		if t.Value.FinishedAt == nil {
+			continue
+		}
+
+		taskAge := now.Sub(t.Value.FinishedAt.Time())
```

Comment:
@michaljurecko Tady bych potřeboval jen potvrdit že jsem to dobře pochopil. Stáří tasku beru z `FinishedAt` a jako úspěšný beru ten co má prázdnou `Error`, ok?

Reply:
Ano.

Reply:
Mozes tam pridat nejake `Is*` metody, aby to bolo aj z kodu jednoznacne.